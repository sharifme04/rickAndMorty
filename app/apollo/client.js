import {ApolloClient, InMemoryCache} from '@apollo/client';
import fetch from 'cross-fetch';
import {createHttpLink} from 'apollo-link-http';

// Initialize Apollo Client
const client = new ApolloClient({
  link: createHttpLink({
    uri: 'https://rickandmortyapi.com/graphql/',
    fetch: fetch,
  }),
  cache: new InMemoryCache(),
});
export default client;
