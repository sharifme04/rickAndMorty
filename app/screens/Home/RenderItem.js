import React from 'react';
import {Text, Image, TouchableOpacity} from 'react-native';
import styles from './Style';

const RenderItem = ({item, navigation}) => (
  <TouchableOpacity
    style={styles.item}
    onPress={() =>
      navigation.navigate('Details', {
        id: item.id,
        image: item.image,
        gender: item.gender,
        species: item.species,
        episode: item.episode,
      })
    }>
    <Image
      style={styles.image}
      source={{
        uri: `${item.image}`,
      }}
    />
    <Text style={styles.title}>{item.name}</Text>
  </TouchableOpacity>
);

export default RenderItem;
