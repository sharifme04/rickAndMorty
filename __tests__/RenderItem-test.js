import 'react-native';
import React from 'react';
import {MockedProvider} from '@apollo/client/testing';
import RenderItem from '../app/screens/Home/RenderItem';
import renderer from 'react-test-renderer';

const item = {
  gender: 'Male',
  id: '1',
  image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
  species: 'Human',
  episode: [
    {air_date: 'December 2, 2013', name: 'Pilot', __typename: 'Episode'},
    {
      air_date: 'December 9, 2013',
      name: 'Lawnmower Dog',
      __typename: 'Episode',
    },
  ],
};

test('RenderItem with item data', () => {
  const tree = renderer
    .create(
      <MockedProvider addTypename={false}>
        <RenderItem item={item} />
      </MockedProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('RenderItem with item empty object ', () => {
  const tree = renderer
    .create(
      <MockedProvider addTypename={false}>
        <RenderItem item={{}} />
      </MockedProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
