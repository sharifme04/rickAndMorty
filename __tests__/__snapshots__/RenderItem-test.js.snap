// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderItem with item data 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "flexDirection": "row",
      "marginVertical": 1,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
      }
    }
    style={
      Object {
        "borderRadius": 25,
        "height": 50,
        "marginRight": 20,
        "width": 50,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  />
</View>
`;

exports[`RenderItem with item empty object  1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "flexDirection": "row",
      "marginVertical": 1,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "undefined",
      }
    }
    style={
      Object {
        "borderRadius": 25,
        "height": 50,
        "marginRight": 20,
        "width": 50,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  />
</View>
`;
